source/Stargate-Model/HTTPBasedRESTfulAPI.class.st

```diff
@@ -2,58 +2,71 @@
 I represent an HTTP based RESTful API, serving one or more services.
 "
 Class {
-	#name : #HTTPBasedRESTfulAPI,
-	#superclass : #Object,
+	#name : 'HTTPBasedRESTfulAPI',
+	#superclass : 'Object',
 	#instVars : [
 		'teapotServer',
 		'controllers',
 		'errorHandlers',
 		'enabledOperationalPlugins',
 		'operationsConfiguration',
-		'disabledOperationalPlugins'
+		'disabledOperationalPlugins',
+		'optionalCORSHandler'
 	],
-	#category : #'Stargate-Model-Controllers'
+	#category : 'Stargate-Model-Controllers',
+	#package : 'Stargate-Model',
+	#tag : 'Controllers'
 }
 
-{ #category : #'instance creation' }
+{ #category : 'instance creation' }
 HTTPBasedRESTfulAPI class >> configuredBy: configuration installing: aRESTfulControllerCollection [
 
 	AssertionChecker enforce: [ aRESTfulControllerCollection notEmpty ] because: 'An API needs at least to expose one resource'.
 	^ self new initializeConfiguredBy: configuration installing: aRESTfulControllerCollection
 ]
 
-{ #category : #'private - configuring' }
+{ #category : 'private - configuring' }
 HTTPBasedRESTfulAPI >> addController: aRESTfulController [
 
 	controllers add: aRESTfulController
 ]
 
-{ #category : #configuring }
+{ #category : 'configuring' }
 HTTPBasedRESTfulAPI >> allowCrossOriginSharingApplying: aConfigurationBlock [
 
 	| builder |
 
-	builder := aConfigurationBlock
-		value: ( CrossOriginResourceSharingFilterBuilder using: controllers ).
+	builder := aConfigurationBlock value:
+		           ( CrossOriginResourceSharingFilterBuilder using: controllers ).
 	builder buildPreflightFilter applyOn: teapotServer.
-	builder buildActualRequestFilter applyOn: teapotServer
+	builder buildActualRequestFilter applyOn: teapotServer.
+	optionalCORSHandler := Optional containing: builder actualRequestHandler
 ]
 
-{ #category : #configuring }
+{ #category : 'private - configuring' }
+HTTPBasedRESTfulAPI >> applyCrossSharingResourceConfigurationTo: response accordingTo: request [
+
+	optionalCORSHandler withContentDo: [ :handler |
```

Comment:
This handler is only used for error handling, right? It would be better to make it explicit?

Reply:
By now it's used in the error handling cases but it's not restricted to that. It's the same handler configured as an after filter for the standard request flow. I don't like the idea to relate it with the error handling per-se, because it can be used to apply the CORS configuration to any request processed out of standard flow.